lib/rust/frp2/src/node/template.rs

```diff
@@ -0,0 +1,358 @@
+    {
+        self._new_node_with_init::<ModelNotUsed, Type, Inputs, Output, F, _Out>(inps, body, init)
+    }
+
+    #[allow(dead_code)]
+    #[inline(always)]
+    fn new_node_with_model_with_init<Type, Inputs, Output, F, _Out>(
+        &self,
+        inps: Inputs,
+        body: F,
+        init: impl FnOnce(&mut NodeData) -> _Out,
+    ) -> NodeInNetwork<Model, TypedNode<Type, Output>>
+    where
+        Inputs: Template<ModelUsed, Model, Output, F>,
+    {
+        self._new_node_with_init::<ModelUsed, Type, Inputs, Output, F, _Out>(inps, body, init)
+    }
+
+    #[inline(always)]
+    pub(crate) fn new_node<Type, Inputs, Output, F>(
+        &self,
+        inps: Inputs,
+        body: F,
+    ) -> NodeInNetwork<Model, TypedNode<Type, Output>>
+    where
+        Inputs: Template<ModelNotUsed, Model, Output, F>,
+    {
+        self.new_node_with_init(inps, body, |_| {})
+    }
+
+    #[inline(always)]
+    pub(crate) fn new_node_with_model<Type, Inputs, Output, F>(
+        &self,
+        inps: Inputs,
+        body: F,
+    ) -> NodeInNetwork<Model, TypedNode<Type, Output>>
+    where
+        Inputs: Template<ModelUsed, Model, Output, F>,
+    {
+        self.new_node_with_model_with_init(inps, body, |_| {})
+    }
+}
+
+
+
+impl<M, Model, Output, F> Template<M, Model, Output, F> for ()
+where
+    M: ModelChooser<Model>,
+    F: Fn(EventContext<Output>, &mut ChosenModel<M, Model>) + 'static,
+{
+    #[inline(always)]
+    fn new_node_with_init<Type>(
+        net: &Network<Model>,
+        _inputs: Self,
+        f: F,
+        init: impl FnOnce(&mut NodeData),
+    ) -> TypedNode<Type, Output> {
+        let model = M::clone_model(&net.model);
+        net.new_node_with_init_unchecked(
+            move |rt: &Runtime, node: &NodeData, _: &dyn Data| {
+                M::with_model_borrow_mut(&model, |model| {
+                    f(EventContext::unchecked_new(rt, node), model)
+                })
+            },
+            init,
+        )
+    }
+}
+
+impl<M, Model, N0, Output, F> Template<M, Model, Output, F> for (Listen<N0>,)
+where
+    M: ModelChooser<Model>,
+    N0: Node,
+    F: 'static + Fn(EventContext<Output>, &mut ChosenModel<M, Model>, &N0::Output),
+{
+    #[inline(always)]
+    #[allow(unsafe_code)]
+    fn new_node_with_init<Type>(
+        net: &Network<Model>,
+        inputs: Self,
+        f: F,
+        init: impl FnOnce(&mut NodeData),
+    ) -> TypedNode<Type, Output> {
+        let model = M::clone_model(&net.model);
+        let inputs = inputs.map_fields_into::<input::Type>();
+        let node = net.new_node_with_init_unchecked(
+            move |rt: &Runtime, node: &NodeData, data: &dyn Data| {
+                #[allow(trivial_casts)]
+                let data = unsafe { &*(data as *const dyn Data as *const N0::Output) };
+                M::with_model_borrow_mut(&model, |model| {
+                    f(EventContext::unchecked_new(rt, node), model, data);
+                })
```

Comment:
This (and other) Template implementations have no explanation why unsafe casts are safe.

Reply:
I added todo there. Explaining it is a little bit hard. I need to spend some time on it. Will be done in the next PR.